test/unit/addressProvider/DAPIAddress.spec.js

```diff
@@ -0,0 +1,237 @@
+const DAPIAddress = require('../../../lib/addressProvider/DAPIAddress');
+const DAPIAddressHostMissingError = require(
+  '../../../lib/addressProvider/errors/DAPIAddressHostMissingError',
+);
+
+describe('DAPIAddress', () => {
+  let dapiAddress;
+
+  describe('#constructor', () => {
+    let host;
+
+    beforeEach(() => {
+      host = '127.0.0.1';
+    });
+
+    it('should construct DAPIAddress from host', () => {
+      dapiAddress = new DAPIAddress(host);
+
+      expect(dapiAddress).to.be.an.instanceOf(DAPIAddress);
+      expect(dapiAddress.host).to.equal(host);
+      expect(dapiAddress.httpPort).to.equal(DAPIAddress.DEFAULT_HTTP_PORT);
+      expect(dapiAddress.grpcPort).to.equal(DAPIAddress.DEFAULT_GRPC_PORT);
+      expect(dapiAddress.proRegTxHash).to.be.undefined();
+      expect(dapiAddress.banCount).to.equal(0);
+      expect(dapiAddress.banStartTime).to.be.undefined();
+    });
+
+    it('should construct DAPIAddress from DAPIAddress', () => {
+      const address = new DAPIAddress(host);
+
+      dapiAddress = new DAPIAddress(address);
+
+      expect(dapiAddress).to.be.an.instanceOf(DAPIAddress);
+      expect(dapiAddress.toJSON()).to.deep.equal(address.toJSON());
+    });
+
+    it('should construct DAPIAddress form RawDAPIAddress', () => {
+      const rawDAPIAddress = new DAPIAddress(host).toJSON();
+
+      dapiAddress = new DAPIAddress(rawDAPIAddress);
+
+      expect(dapiAddress).to.be.an.instanceOf(DAPIAddress);
+      expect(dapiAddress.host).to.equal(host);
+      expect(dapiAddress.httpPort).to.equal(DAPIAddress.DEFAULT_HTTP_PORT);
+      expect(dapiAddress.grpcPort).to.equal(DAPIAddress.DEFAULT_GRPC_PORT);
+      expect(dapiAddress.proRegTxHash).to.be.undefined();
+      expect(dapiAddress.banCount).to.equal(0);
+      expect(dapiAddress.banStartTime).to.be.undefined();
+    });
+
+    it('should construct DAPIAddress with defined ports', () => {
+      const httpPort = DAPIAddress.DEFAULT_HTTP_PORT + 1;
+      const grpcPort = DAPIAddress.DEFAULT_GRPC_PORT + 1;
+      const proRegTxHash = 'proRegTxHash';
+
+      dapiAddress = new DAPIAddress({
+        host,
+        httpPort,
+        grpcPort,
+        proRegTxHash,
+        banCount: 100,
```

Comment:
`banCount` and `banStartTime` not part of `RawDAPIAddress`, let's remove them from here

Reply:
the idea was to check that we don't set these values

Reply:
I see. In this case, I would move them into a separate test case like "should not set banCount and banStartTime from RawDAPIAddress" in order to avoid confusion.